lib/decompress.ps1

```diff
@@ -1,123 +1,121 @@
 function requires_7zip($manifest, $architecture) {
-    foreach($dlurl in @(url $manifest $architecture)) {
-        if(file_requires_7zip $dlurl) { return $true }
+    foreach ($dlurl in @(url $manifest $architecture)) {
```

Comment:
Why do not go with something like:

```suggestion
    return (@(url $manifest $architecture) | Where-Object { file_requires_7zip $_ }).Count -gt 0
```

![powershell](https://i.imgur.com/M4fO3T3.png)

Reply:
Adopt suggestion in `requires_7zip` and `requires_msi`